scripts/pi-hole/php/teleporter.php

```diff
@@ -118,40 +106,43 @@ function getWildcardListContent() {
 			}
 		}
 
-		$continue = strtolower($name[1]) == 'zip' ? true : false;
+		$continue = strtolower($name[1]) == 'tar' && strtolower($name[2]) == 'gz' ? true : false;
 		if(!$continue || !$okay) {
-			die("The file you are trying to upload is not a .zip file (filename: ".$filename.", type: ".$type."). Please try again.");
+			die("The file you are trying to upload is not a .tar.gz file (filename: ".$filename.", type: ".$type."). Please try again.");
 		}
 
-		$zip = new ZipArchive();
-		$x = $zip->open($source);
-		if ($x === true) {
-			if(isset($_POST["blacklist"]))
+		$filename = "/tmp/".$filename;
+		if(!move_uploaded_file($_FILES["zip_file"]["tmp_name"], $filename))
```

Comment:
`$source` has been assigned to `$_FILES["zip_file"]["tmp_name"]`, so you can use the former for conciseness (and below as well)